### PR TITLE
Added baseFeePerGas field in API return

### PR DESCRIPTION
**Pull request by jimni1222**
## Proposed changes

This PR introduces adding `baseFeePerGas` field in `klay_getHeaderByHash` and `klay_getHeaderByNumber`.

To test this, i used the corecell project and enabled `ethTxTypeCompatibleBlock` in `genesis.json` file.

```
> klay.getHeaderByNumber('latest')
{
  baseFeePerGas: "0x0",
  blockScore: "0x1",
  extraData: "0xda830...",
  gasUsed: "0x0",
  governanceData: "0x",
  hash: "0x4f417b2a72c9bd0d0fa67da900340a2e170633d88945e0b949fb8b0bbdf83c3c",
  logsBloom: "0x00000...",
  number: "0x13",
  parentHash: "0xb940e9974b38d8c2ba13e6ac340d3ac5e246aaccdfa7ae9adec17f0a7a5106cc",
  receiptsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  reward: "0x4b2c736fd05c2e2da3ccbd001a395a444f16a861",
  stateRoot: "0xf2d7d66ac2a235945a43570d78cd382d7a812f452deceb08a2bbebbde554c34f",
  timestamp: "0x620ca3fa",
  timestampFoS: "0x3b",
  transactionsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
}
```

```
> klay.getBlock('latest')
{
  baseFeePerGas: "0x0",
  blockscore: "0x1",
  extraData: "0xda830...",
  gasUsed: 0,
  governanceData: "0x",
  hash: "0x3aaf61ce21fcd9aebfca9778117b1641826920f7e5db0171fd1f512d6fe144be",
  logsBloom: "0x00000...",
  number: 40,
  parentHash: "0xd6822d4b1403bff1e8162aa91ee0b265d8c2df8b4c07593e43f454b23bf65ee1",
  receiptsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  reward: "0x4b2c736fd05c2e2da3ccbd001a395a444f16a861",
  size: 624,
  stateRoot: "0xb64deb56ba72af49621829b43c63939d5c69413579b1581b6a6a984a2643b805",
  timestamp: 1645406919,
  timestampFoS: 6,
  totalBlockScore: "0x29",
  transactions: [],
  transactionsRoot: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  voteData: "0x"
}
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: api/api_public_blockchain.go [hunk 1]
```diff
@@ -150,14 +150,58 @@ func (s *PublicBlockChainAPI) GetAccount(ctx context.Context, address common.Add
...
+		"governanceData":   hexutil.Bytes(head.Governance),
+		"hash":             head.Hash(),
+	}
+
+	if len(head.Vote) != 0 {
+		result["voteData"] = hexutil.Bytes(head.Vote)
+	}
+
+	if isEnabledEthTxTypeFork {
+		result["baseFeePerGas"] = hexutil.Uint64(params.BaseFee)
```
Comment (sirano11) [inline api/api_public_blockchain.go:183]:
```suggestion
		result["baseFeePerGas"] = (*hexutil.Big)(params.BaseFee)
```

Comment (jimni1222) [reply api/api_public_blockchain.go]: I modified type to *hexutil.Big in https://github.com/klaytn/klaytn/pull/1183/commits/475ba87330f9b9a11840608abb7b5f95972777e5

**Review by aeharvlee — DISMISSED** [review]
LGTM :)

**Review by sirano11 — DISMISSED** [review]

File: api/api_public_blockchain.go [hunk 2]
```diff
@@ -150,14 +150,59 @@ func (s *PublicBlockChainAPI) GetAccount(ctx context.Context, address common.Add
...
 
+// rpcMarshalHeader converts the given header to the RPC output.
+func (s *PublicBlockChainAPI) rpcMarshalHeader(header *types.Header) map[string]interface{} {
+	fields := RPCMarshalHeader(header, s.b.ChainConfig().IsEthTxTypeForkEnabled(header.Number))
+	return fields
+}
+
+// RPCMarshalHeader converts the given header to the RPC output that includes the baseFeePerGas field.
+func RPCMarshalHeader(head *types.Header, isEnabledEthTxTypeFork bool) map[string]interface{} {
+	fmt.Printf("head.TimeFoS: %v / hexutil.Uint(head.TimeFoS): %v / hexutil.Uint64(head.TimeFoS): %v \n", head.TimeFoS, hexutil.Uint(head.TimeFoS), hexutil.Uint64(head.TimeFoS))
```
Comment (aidan-kwon) [inline api/api_public_blockchain.go:161]:
```suggestion
```

**Review by aidan-kwon — COMMENTED** [review]
Don't we need to add `baseFee` fields in `GetBlock` APIs?

**Comment by jimni1222** [comment]
@aidan-kwon 

> Don't we need to add `baseFee` fields in `GetBlock` APIs?

I will add that. Thank you.

**Comment by jimni1222** [comment]
@aidan-kwon @sirano11 @aeharvlee I added `baseFeePerGas` in GetBlock RPC call. Please take a look :)

**Review by aeharvlee — APPROVED** [review]

**Comment by KimKyungup** [comment]
@jimni1222 Like our `EthereumAPI`, don't we need to add `rpcMarshalHeader` in `func (api *PublicFilterAPI) NewHeads(`?

I refered below code.
```
// NewHeads send a notification each time a new (header) block is appended to the chain.
func (api *EthereumAPI) NewHeads(ctx context.Context) (*rpc.Subscription, error) {
	notifier, supported := rpc.NotifierFromContext(ctx)
	if !supported {
		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
	}

	rpcSub := notifier.CreateSubscription()
	go func() {
		headers := make(chan *types.Header)
		headersSub := api.publicFilterAPI.Events().SubscribeNewHeads(headers)

		for {
			select {
			case h := <-headers:
				header, err := api.rpcMarshalHeader(h)
				if err != nil {
					logger.Error("Failed to marshal header during newHeads subscription", "err", err)
					headersSub.Unsubscribe()
					return
				}
				notifier.Notify(rpcSub.ID, header)
```

**Comment by jimni1222** [comment]
> @jimni1222 Like our `EthereumAPI`, don't we need to add `rpcMarshalHeader` in `func (api *PublicFilterAPI) NewHeads(`?
> 
> I refered below code.
> 
> ```
> // NewHeads send a notification each time a new (header) block is appended to the chain.
> func (api *EthereumAPI) NewHeads(ctx context.Context) (*rpc.Subscription, error) {
> 	notifier, supported := rpc.NotifierFromContext(ctx)
> 	if !supported {
> 		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
> 	}
> 
> 	rpcSub := notifier.CreateSubscription()
> 	go func() {
> 		headers := make(chan *types.Header)
> 		headersSub := api.publicFilterAPI.Events().SubscribeNewHeads(headers)
> 
> 		for {
> 			select {
> 			case h := <-headers:
> 				header, err := api.rpcMarshalHeader(h)
> 				if err != nil {
> 					logger.Error("Failed to marshal header during newHeads subscription", "err", err)
> 					headersSub.Unsubscribe()
> 					return
> 				}
> 				notifier.Notify(rpcSub.ID, header)
> ```

Thank you for your review. I put `RPCMarshalHeader` function in the `filters` package, and now `api` package uses that function.
And tested with wscat like below. You can check `baseFeePerGas` field.

```
> wscat -c ws://localhost:8552
> {"jsonrpc":"2.0", "id": 1, "method": "klay_subscribe", "params": ["newHeads"]}
< {"jsonrpc":"2.0","id":1,"result":"0xc2719323c5bdf897030dd6db05684f26"}
< {"jsonrpc":"2.0","method":"klay_subscription","params":{"subscription":"0xc2719323c5bdf897030dd6db05684f26","result":{"baseFeePerGas":"0x0","blockScore":"0x1","extraData":"0xda830...","gasUsed":"0x0","governanceData":"0x","hash":"0xb51508cfbef4b561224d06d7fd25aa8fc3aafbb2ba48600f5aaabf7c4e4de1c4","logsBloom":"0x00000...","number":"0x12b","parentHash":"0x39a49d4c45593976651cbbc1f602652eb3b483236917506dd8f62c03ccb98d65","receiptsRoot":"0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470","reward":"0x4b2c736fd05c2e2da3ccbd001a395a444f16a861","stateRoot":"0xad45c4364c54dce19bf6de6bb1be5ee5aaa8ef6d8aa8b32d27f59ddcecf17916","timestamp":"0x621349a1","timestampFoS":"0x0","transactionsRoot":"0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"}}}
```

**Review by KimKyungup — APPROVED** [review]
LGTM

**Review by aidan-kwon — APPROVED** [review]
LGTM Thanks!

**Review by aeharvlee — APPROVED** [review]

**Pull request merged**